weave/compile.py

```diff
@@ -89,16 +89,13 @@ def _call_execute(function_node: graph.Node) -> graph.OutputNode:
     )
 
 
-def _quote_node(node: graph.Node, into_node: graph.OutputNode) -> graph.Node:
-    # Special case for ops that want to preserve the exact user-defined node chain
-    from .panels_py import generator_templates
-
-    compile_time_literal = into_node.from_op.name in [
-        spec.op_name
-        for spec in generator_templates.template_registry.get_specs().values()
-    ]
-
-    return weave_internal.const(node, _compile_time_literal=compile_time_literal)
+def _quote_node(node: graph.Node) -> graph.Node:
```

Comment:
_quote_node is not called in the case where the has already wrapped the passed in node in a const themselves. But I think in the case where the user has done, we still want to compile the quoted node so we can swap to the refined output type.

So I think you'll need to change the compile_quote pass entirely. Or you add a new pass after compile_quote that maps through looking for any arg that accepts FunctionType (with no input_types) and call compile on the node in that position, swapping its type. A new pass for this probably makes sense.

Reply:
Fixed in next commit